README.md

```diff
@@ -1 +1,409 @@
+ * {@arg source Test1}
+ */
+private interface Test1
+
+/**
+ * Hello World 2!
+ * @include [Test1]
+ * {@arg source Test2}
+ */
+@AnnotationTest(a = 24)
+private interface Test2
+
+/** 
+ * Some extra text
+ * @include [Test2]
+ * {@arg source someFun} */
+fun someFun() {
+    println("Hello World!")
+}
+
+/** {@include [com.example.plugin.Test2]}{@arg source someMoreFun} */
+fun someMoreFun() {
+    println("Hello World!")
+}
+```
+turns into:
+```kotlin
+package com.example.plugin
+
+/**
+ * Hello World!
+ *
+ * This is a large example of how the plugin will work from Test1
+ *
+ * @param name The name of the person to greet
+ * @see [com.example.plugin.KdocIncludePlugin]
+ *
+ */
+private interface Test1
+
+/**
+ * Hello World 2!
+ * Hello World!
+ *
+ * This is a large example of how the plugin will work from Test2
+ *
+ * @param name The name of the person to greet
+ * @see [com.example.plugin.KdocIncludePlugin][com.example.plugin.KdocIncludePlugin]
+ *
+ */
+@AnnotationTest(a = 24)
+private interface Test2
+
+/**
+ * Some extra text
+ * Hello World 2!
+ * Hello World!
+ *
+ * This is a large example of how the plugin will work from someFun
+ *
+ * @param name The name of the person to greet
+ * @see [com.example.plugin.KdocIncludePlugin][com.example.plugin.KdocIncludePlugin]
+ */
+fun someFun() {
+   println("Hello World!")
+}
+
+/** Hello World 2!
+ * Hello World!
+ *
+ * This is a large example of how the plugin will work from someMoreFun
+ *
+ * @param name The name of the person to greet
+ * @see [com.example.plugin.KdocIncludePlugin][com.example.plugin.KdocIncludePlugin]
+ */
+fun someMoreFun() {
+   println("Hello World!")
+}
+```
+
+## Technicalities
+
+KDocs and JavaDocs are structured in a tree-like structure and are thus also parsed and processed like that.
+For example, the following KDoc:
+```kotlin
+/** 
+ * Some extra text
+ * @a [Test2]
+ * Hi there!
+ * @b source someFun
+ * Some more text. (
+ * @c [Test1] {
+ * )
+ */
+```
+will be parsed as follows:
+```
+["
+Some extra text",
+"@a [Test2]
+Hi there!",
+"@b source someFun
+Some more text. (
+@c [Test1] {
+)
+"]
+```
+
+This is also how tag processors receive their data. Note that any newlines after the `@tag`
+are also included as part of the tag data. Tag processors can then decide what to do with this extra data.
+However, `@include`, `@includeArg`, `@sample`, and `@includeFile` all have systems in place that
+will keep the content after the tag and on the lines below the tag in place. 
+Take this into account when writing your own processors.
+
+To avoid any confusion, it's usually easier to stick to `{@inline tags}` as then it's clear which part of the doc
+belongs to the tag and what does not. Inline tags are processed before block tags.
+
+Take extra care when using tags that can introduce new tags, such as `@include`, as this will cause the structure
+of the doc to change mid-processing. Very powerful, but also potentially dangerous.
```

Comment:
No, the plugin will not crash (infinite loops also produce nice and helpful errors), but it might be hard for the user to understand what's going on. Like if they do:
```
Test
@comment Hi
@include [Something]
```
=== Include processor =>
```
Test
@comment Hi
Some documentation
```
=== Comment processor =>
```
Test
```